src/Concerns/InteractsWithSourceFiles.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace BeyondCode\QueryDetector\Concerns;
+
+trait InteractsWithSourceFiles
```

Comment:
The methods are not used anywhere except for the QueryDetector, so a trait does not make a lot of sense to me.